flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/adaptive/ExecutingTest.java

```diff
@@ -570,8 +659,9 @@ public void setHowToHandleFailure(Function<Throwable, FailureResult> function) {
             this.howToHandleFailure = function;
         }
 
-        public void setCanScaleUp(Supplier<Boolean> supplier) {
-            this.canScaleUp = supplier;
+        public void setCanScaleUp(Boolean minIncreaseMet, Boolean parallelismChangeAfterTimeout) {
+            this.minIncreaseMet = minIncreaseMet;
+            this.parallelismChangeAfterTimeout = parallelismChangeAfterTimeout;
```

Comment:
Same comment with prod comment[1], Executing side just care about whether `forceRescale` is ture or false, and shouldn't mention minIncreaseMet or parallelismChangeAfterTimeout.

That's why the old code is `canScaleUp` instead of `minIncreaseMet`. I prefer rename them to `canScaleUpWithoutForce` and `canScaleUpWithForce`.

Also, these 2 options can be set by 2 setter methods. 

WDYT?

[1] https://github.com/apache/flink/pull/22985#discussion_r1356382291

Reply:
Agree for the renaming. But regarding the setters I disagree because the 2 parameters are always set together: it is their combination that gives the output of the test. I find it way clearer to set them together.